neuro_flow/batch_runner.py

```diff
@@ -296,31 +296,21 @@ async def process(
                         ):
                             topos[new_prefix] = (new_ctx, new_topo)
 
-                await self._process_started(attempt, topos, started, finished, skipped)
+                ok = await self._process_started(
+                    attempt, topos, started, finished, skipped
+                )
+                if not ok:
+                    await self._do_cancellation(
+                        attempt, topos, started, finished, skipped
+                    )
+                    return
```

Comment:
Currently, when just before any exit of `.process()`, we print something like "Attempt #1 completed". I think it will be good to keep it this way.

Reply:
You are right.
Replaced `return` with `break` to finalize an attempt in a normal way.